comp/core/hostname/hostnameimpl/service.go

```diff
@@ -43,7 +43,18 @@ func (hs *service) GetWithProvider(ctx context.Context) (pkghostname.Data, error
 	return pkghostname.GetWithProvider(ctx)
 }
 
-// newHostnameService fetches the hostname and returns a service wrapping it
+// TODO: (components) remove once compliance, epforwarder, and security are FX components
+//
+//nolint:revive // TODO(AML) Fix revive linter
+func NewHostnameService() hostname.Component {
+	return newHostnameServiceCompat()
+}
+
+// TODO: (components) - merge with newHostnameServiceCompat once NewHostnameService is removed
 func newHostnameService() hostname.Component {
+	return newHostnameServiceCompat()
+}
+
+func newHostnameServiceCompat() hostname.Component {
```

Comment:
Can you explain why there are 3 constructors and not just one?

Reply:
This was done in order to work around the private constructor for non-fx apps, see comment on original PR for reference: https://github.com/DataDog/datadog-agent/pull/21987#discussion_r1450897502

Reply:
What do you think about having only `NewHostnameService()`?

Reply:
That would be fine with me, but I don't have as much experience with the agent codebase so I'd defer to you or @gh123man

Reply:
@ogaca-dd are we ok with having public constructors in the component implementation? I am worried this will cause issues down the road (for components that depend on other components).

Reply:
In an ideal world we shouldn't have exported constructor but I think this is OK in this case: https://uber-go.github.io/fx/modules.html#export-boundary-functions

Reply:
Ok great - thanks for the link, this provides clear guidance. 
@liustanley feel free to ignore my previous feedback and take @ogaca-dd 's suggestions. 🚀

Reply:
Thanks! Just updated the constructors